src/aeso_ast_to_icode.erl

```diff
@@ -17,11 +17,15 @@
 
 -spec convert_typed(aeso_syntax:ast(), list()) -> aeso_icode:icode().
 convert_typed(TypedTree, Options) ->
-    Name = case lists:last(TypedTree) of
-               {contract, _, {con, _, Con}, _} -> Con;
-               _ -> gen_error(last_declaration_must_be_contract)
+    {Payable, Name} =
+        case lists:last(TypedTree) of
+            {contract, Attrs, {con, _, Con}, _} ->
+                {proplists:get_value(payable, Attrs, false), Con};
+            _ ->
+                gen_error(last_declaration_must_be_contract)
            end,
-    NewIcode = aeso_icode:set_name(Name, aeso_icode:new(Options)),
+    NewIcode = aeso_icode:set_payable(Payable,
```

Comment:
So here, `Payable` is definitely a boolean, but in `aeso_icode:set_payable/2`, the first argument is called `Name` (though typed as a boolean), which at least looking at this context, seems confusing.